python/video_creator/video_creator.py

```
@@ -0,0 +1,310 @@
+        # False when EOF
+        if not success:
+            break
+        batch.append({'data': cv2.imencode('.jpg', image)[1].tostring(), 'custom_id': 'Frame_{}'.format(frame_number)})
+        image_count += 1
+        frame_number += 1
+    return batch, frame_number, success
+
+
+def get_results(client, feed_id, classes, video, batch_size, max_threads):
+    '''
+    Creates an array of dicts that store results. Each dict corresponds to an image, and has a key corresponding to each class.
+    '''
+    frame_number = 0
+    success = True
+    result_batches = []
+
+    current_batch_number = 0
+    total_batches = int(video.get(cv2.CAP_PROP_FRAME_COUNT) // batch_size)
+    with ThreadPool(max_threads) as pool:
+        while success:
+            async_responses = []
+            thread_count = 0
+            # Process max_threads number of image batches at a time
+            while thread_count < max_threads and success:
+                print('Processing batch {}/{}.'.format(current_batch_number, total_batches))
+                image_blob_batch, frame_number, success = get_image_blob_batch(video, batch_size, frame_number)
+                # Get response object for each async function call
+                if len(image_blob_batch) > 0:
+                    try:
+                        async_responses.append(pool.apply_async(run_classification, (client, feed_id, classes, image_blob_batch)))
+                        thread_count += 1
+                        current_batch_number += 1
+                    except Exception as e:
+                        # FIXME
+                        logging.warning('Ava Exception: {}'.format(str(e)))
+                        success = False
+            for response in async_responses:
+                result_batches.append(response.get())
+            if current_batch_number == 50:
+                success = False
+    # Remove batching and add all results to a single list
+    results = []
```

I'd consider refactoring this part too with the intention of reducing indentation (you have 3 nested for loops!). I'd also remove the list comprehension below. They can get pretty gnarly to read when it's so long. 

Also, **never** use `assert` in your code. Only use it in unit tests. What you'll get are `AssertionError` exceptions and they're basically useless. If you need to throw an exception then try to throw one that makes more sense. I might consider a custom exception here but perhaps `ValueError` might work too.

```python
if len(results) == frame_number:
  raise ValueError('nice readably error message')
```